core/src/datayoga_core/write_utils.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from itertools import groupby
+from typing import Any, Callable, Dict, List, Union
+
+from datayoga_core.block import Block
+from datayoga_core.opcode import OpCode
+from datayoga_core.result import Result, Status
+
+logger = logging.getLogger("dy")
+
+
+def write_records(data: List[Dict[str, Any]],
```

Comment:
I would prefer not to send Callables as method parameters. It can get very messy very quickly.
Why not have this method just do the batching for you, and return 3 batches. The block can then decide what to do with it. We can then implement the deduplication here too.

Reply:
done